nixos/modules/security/google_oslogin.nix

```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.security.googleOSLogin;
+  package = pkgs.google-compute-engine-oslogin;
+
+in
+
+{
+
+  options = {
+
+    security.googleOSLogin.enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable Google OS Login
+
+        The OS Login package enables the following components:
+        AuthorizedKeysCommand to query valid SSH keys from the user's OS Login
+        profile during ssh authentication phase.
+        NSS Module to provide user and group information
+        PAM Module providing authorization and authentication support, allowing
+        the system to use data stored in Google Cloud IAM permissions to
+        control both, the ability to log into an instance, and to perform
+        operations as root (sudo).
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+    security.pam.enableGoogleOSLogin = true;
+
+    security.sudo.extraConfig = ''
+      #includedir /run/google-sudoers.d
+    '';
+    systemd.tmpfiles.rules = [
+      "d /run/google-sudoers.d 660 root root -"
```

Comment:
The `-` for age means systemd will create the directory with mentioned permissions and ownership, but not automatically clean up anything.

I moved that from `/var/google-sudoers.d` (and patched `pam_module/pam_oslogin_admin.cc`), as that location seemed much more suitable for runtime files.

`pam_oslogin_admin.cc` has code to clean up existing files if admin status is revoked, it's just a more meaningful location (and cleans up expired users on a reboot)